### ISSUE TITLE
Tell a point with its uncertainty

### ISSUE THREAD

**Issue by glouppe** [scikit-optimize/scikit-optimize#300]
In the long term, it would be nice if one could `tell` a pair `(x, y)` along with uncertainty on `y`.

For example, when `y` is a cross-validation estimate of the performance of a ML model parameterized by `x`, then we also have an uncertainty estimate of `y`. This piece of information is currently not leveraged, which is too bad!

Not that this is however not a trivial change, since none of the regression models we have support that. The old version of gaussian processes in scikit-learn supported a `nugget` argument though, through which this information could be fed.

**Comment by MechCoder** [comment]
`alpha` does the same work as what `nugget` does no? (it can be array as well). I opened an issue here (https://github.com/scikit-learn/scikit-learn/issues/7821) to set it at fit time but was unable to review the Pull Request.

**Comment by glouppe** [comment]
Good point. While I am not sure about what will happen with regards to adding `sample_alpha` at `fit` (see scikit-learn/scikit-learn#7985), we could backport such change in skopt.
